train.py

```diff
@@ -77,7 +82,71 @@ def get_base_argument_parser():
     return parser
 
 
-############################################################################## Training
+def housekeeping():
+    argument_parser = get_base_argument_parser()
+    args = process_args(argument_parser)
+
+    if args.gpu_ids_to_use is None:
+        select_devices(
+            args.num_gpus_to_use,
+            max_load=args.max_gpu_selection_load,
+            max_memory=args.max_gpu_selection_memory,
+            exclude_gpu_ids=args.excude_gpu_list,
+        )
+    else:
+        os.environ["CUDA_VISIBLE_DEVICES"] = args.gpu_ids_to_use.replace(" ", ",")
+
+    saved_models_filepath, logs_filepath, images_filepath = build_experiment_folder(
+        experiment_name=args.experiment_name, log_path=args.logs_path
+    )
+
+    args.saved_models_filepath = saved_models_filepath
+    args.logs_filepath = logs_filepath
+    args.images_filepath = images_filepath
+
+    #  if you have an environment variable set, prioritise this over the default
+    # argparse option
```

Comment:
I don't agree with this. Wouldn't the proper way to do this be:

1. Environment variable read.
2. If argument for path is not None, read that in, otherwise keep the environment variable as the path.

Reply:
Would agree with this, but it will require us to have the default data path set to None. 

I'm happy with that but can understand that others might not be.

Reply:
I guess I don't see why we even need the environment variable in the first place. Couldn't one simply pass the environment variable as the argument to data path?

Reply:
This was a personal preference from my end, but I can see maybe it's too opinionated. 

The reason I have it set is so that I can have one config file that I know can run on all of the machines (including my laptop), even if the data is stored in different places. 

This way when I run `python train.py -config abc.json`, it doesn't matter if the data is in scratch_fast, scratch_ssd, ~/datasets, etc. the script will work out of the box. I find it a nice convenience, but if no one else is going to use it then I'd be happy for it to be removed.

Reply:
Can't you pass an environment variable in the config? We can check if the argument is a path or an environment variable by some unique identifier.

Reply:
That way you can get what you want, without requiring explicit handling of environment variables internally. If the path is None we throw an error.

Reply:
Oh yeah nice idea. Sounds good

Reply:
Alright. Should we use the $VARIABLE identifier?